tests/expose_test.go

```diff
@@ -0,0 +1,384 @@
+package tests_test
+
+import (
+	"flag"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/google/goexpect"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	k8sv1 "k8s.io/api/core/v1"
+	k8smetav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	v1 "kubevirt.io/kubevirt/pkg/api/v1"
+	"kubevirt.io/kubevirt/pkg/kubecli"
+	"kubevirt.io/kubevirt/pkg/virtctl/expose"
+	"kubevirt.io/kubevirt/pkg/virtctl/offlinevm"
+	"kubevirt.io/kubevirt/tests"
+)
+
+var _ = Describe("Expose", func() {
+
+	flag.Parse()
+
+	virtClient, err := kubecli.GetKubevirtClient()
+	tests.PanicOnError(err)
+	const testPort = 1500
+
+	Context("Expose service on a VM", func() {
+		By("Creating a VM object")
+		vm := tests.NewRandomVMWithEphemeralDiskAndUserdata(tests.RegistryDiskFor(tests.RegistryDiskCirros), "#!/bin/bash\necho 'hello'\n")
+		vm.Labels = map[string]string{"expose": "vm"}
+
+		It("Should start a VM and run a 'hello world' server on it", func() {
+			By("Starting the VM")
+			vm, err = virtClient.VM(tests.NamespaceTestDefault).Create(vm)
+			Expect(err).ToNot(HaveOccurred())
+			tests.WaitForSuccessfulVMStartIgnoreWarnings(vm)
+
+			By("Making sure the VM could be fetched")
+			vm, err = virtClient.VM(tests.NamespaceTestDefault).Get(vm.ObjectMeta.Name, k8smetav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Making sure that the OS is up and ready for login")
+			_, err := tests.LoggedInCirrosExpecter(vm)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Running a 'hello world' server on the VM")
+			expecter, _, err := tests.NewConsoleExpecter(virtClient, vm, 10*time.Second)
+			defer expecter.Close()
+			Expect(err).ToNot(HaveOccurred())
+			_, err = expecter.ExpectBatch([]expect.Batcher{
+				&expect.BSnd{S: "\n"},
+				&expect.BExp{R: "\\$ "},
+				&expect.BSnd{S: fmt.Sprintf("screen -d -m nc -klp %d -e echo -e \"Hello World!\"\n", testPort)},
+				&expect.BExp{R: "\\$ "},
+				&expect.BSnd{S: "echo $?\n"},
+				&expect.BExp{R: "0"},
+			}, 60*time.Second)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		Context("Expose ClusterIP service", func() {
+			const servicePort = "27017"
+			const serviceName = "cluster-ip-vm"
+			It("Should expose a Cluster IP service on a VM", func() {
```

Comment:
seems like i have the same issue here. if the test in line 75 happens before this, then it would fail. will merge to one "It" section